napari_brightness_contrast/_dock_widget.py

```diff
@@ -321,24 +314,27 @@ def histogram(layer, num_bins : int = 256, minimum = None, maximum = None, use_c
     # delete cache when data is changed
     def _refresh_data(event):
         reset_histogram_cache(layer)
-
         layer.events.data.disconnect(_refresh_data)
+
     layer.events.data.connect(_refresh_data)
+   
 
     layer.metadata["bc_histogram_num_bins"] = num_bins
     layer.metadata["bc_histogram"] = hist
 
     return hist
 
 def reset_histogram_cache(layer):
-    if "bc_histogram_num_bins" in layer.metadata.keys() and "bc_histogram" in layer.metadata.keys():
+    try:
         layer.metadata.pop("bc_histogram_num_bins")
         layer.metadata.pop("bc_histogram")
+    except KeyError:
+        return
+    return
 
 def min_max(data):
-    if "dask" in str(type(data)): # ugh
-        data = np.asarray(data)
-
+    data = np.asarray(data)
```

Comment:
Can we revert your change here? This change makes the code slower in case the image lives in GPU memory (pyopencl for example, and in the future also with cupy)...
```suggestion
    if "dask" in str(type(data)): # ugh
        data = np.asarray(data)
```

Reply:
Sure, this make sense